pkg/apply/solver/solver.go

```diff
@@ -211,26 +211,22 @@ func (t *TaskQueueBuilder) Build(taskContext *taskrunner.TaskContext, o Options)
 		}
 	}
 
-	// TODO: add InvSetTask when Destroy=true to retain undeleted objects
-	if !o.Destroy {
-		klog.V(2).Infoln("adding inventory set task")
-		prevInvIds, _ := t.InvClient.GetClusterObjs(t.invInfo)
-		tasks = append(tasks, &task.InvSetTask{
-			TaskName:      "inventory-set-0",
-			InvClient:     t.InvClient,
-			InvInfo:       t.invInfo,
-			PrevInventory: prevInvIds,
-			DryRun:        o.DryRunStrategy,
-		})
+	prevInvIds, _ := t.InvClient.GetClusterObjs(t.invInfo)
+	klog.V(2).Infoln("adding delete/update inventory task")
+	var taskName string
+	if o.Destroy {
+		taskName = "delete-inventory-0"
```

Comment:
lets change the test name of this task too. It will require some find/replace, but I want to make it clear that it's not always a delete, because it shows up in logs. how about "inventory-delete-or-update-0"?

Reply:
done